src/libraries/Microsoft.Extensions.Logging.Console/src/Formatter/IConsoleLogFormatter.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.Extensions.Logging.Console
+{
+    /// <summary>
+    /// Allows custom log messages formatting
+    /// </summary>
+    public interface IConsoleLogFormatter
+    {
+        /// <summary>
+        /// Gets the name associated with the console log formatter.
+        /// </summary>
+        string Name { get; }
+
+        /// <summary>
+        /// Formats a log message at the specified log level.
```

Comment:
- [ ] add proper comment on here explaining that the formatter would log by embedding ANSI color codes and parse them out upon writing to console